File: backend/weave-java/src/main/java/almanac/models/person/Person.java
package almanac.models.person;

public class Person {
    private Long id;
    private String name;
    private String race;
    private String description;

    public Person(Long id, String name, String race, String description) {
        this.id = id;
        this.name = name;
        this.race = race;
        this.description = description;
    }

    public Long getId() {
        return id;
    }

    public String getName() {
        return name;
    }

    public String getRace() {
        return race;
    }

    public String getDescription() {
        return description;
    }
}

File: Weave/src/main/kotlin/com/api/weave/models/Person.kt
package com.api.weave.models

data class Person(
        val id: Long,
        val name: String,
        val race: String, //enum?
        val description: String
)

data class PersonPage(
        val person: Person
)
data class PersonRelation(
        val id: Long,
        val relatedId: Long,
        val relation: PersonRelationType
)

enum class PersonRelationType {
    DENIZEN,
    OWNER,
    WORKER,
    MEMBER,
}
